### ISSUE TITLE
Pressure Outliers in PIMD NPT Simulations Using i-PI

### ISSUE THREAD

**Issue by jorgecastro316** [i-pi/i-pi#398]
Dear all,

I've encountered an issue while equilibrating liquid water simulations using the qTIP4P/F model in an NPT ensemble. I'm using the isotropic and sc-isotropic barostats in i-PI, with τ values of 5 and 10 fs for both. For volume control, I use a Langevin thermostat with τ = 5 fs, and a PILE-L thermostat with τ = 10 fs for temperature control.

Temperature and volume equilibrate well, reaching Gaussian distributions as expected. However, I'm seeing irregular behavior in the pressure — specifically, there are random, high-value outliers during a single time step, where the pressure significantly deviates from the normal range. The pressure values before and after these outliers are normal, and I haven't noticed any discontinuities in the other calculated properties. Although the average pressure meets the target value, these outliers prevent the pressure distribution from forming a proper Gaussian profile.

Interestingly, this only happens when multiple beads are used in the simulation. When I run classical NPT molecular dynamics, the system equilibrates without issues. I also tried recalculating the pressure for the problematic configurations in replay mode, and the pressure was more consistent with expected values, suggesting that this might be related to the i-PI driver rather than the configurations.

I've attached a screenshot of the input file that corresponds to an equilibration using a partially adiabatic elevated temperature ansatz CMD. The same issue also occurs when using standard PIMD to equilibrate the NPT ensemble. The initial chk file was obtained from a equilibrated classical MD simulation. 

If anyone has suggestions or ideas on what might be going wrong, I'd really appreciate it!

Thank you for your help.

Best regards,
Jorge Castro
PhD student.
MPI for the Structure and Dynamics of Matter
![Screenshot from 2024-11-12 09-28-17](https://github.com/user-attachments/assets/6226efb4-6dae-45c3-98c8-35bf977e5451)

**Comment by venkatkapil24** [comment]
Hi Jorge, it would be useful to see the pressure time series and the distribution. For small systems, it is not uncommon for pressure to have very high values. Another thing that I have noticed is that the tau for the barostat is very small. I would generally prescribe 100 fs for an aggressive barostat and 1000 fs for a gentle one.

**Comment by venkatkapil24** [comment]
Also, the sc-isotropic is not appropriate as it works with the Suzuki-Chin ensemble. Do you see the same issue with the standard isotropic barostat?

**Comment by jorgecastro316** [comment]
I had some doubts about using the SC-isotropic barostat, but since the i-PI documentation recommends it for liquids, I decided to give it a shot. Attached are the input files I used for the simulations at 300 K and 0.40 GPa. The system is composed of 64 water molecules. Unfortunately, I observe the same pressure outlier issue even when using the standard barostat.

Below, I've included the pressure time series and the corresponding histogram of the pressure distribution. The mean value is around 0.40 GPa, as expected, but the outliers persist.

[press_issue.zip](https://github.com/user-attachments/files/17715496/press_issue.zip)
![Screenshot from 2024-11-12 12-06-55](https://github.com/user-attachments/assets/f380aa12-2bd3-45fc-b313-ebd3e7fe0f33)

**Comment by venkatkapil24** [comment]
The SC barostat uses a finite difference, which could be problematic with noise forces. However, I am surprised that the documentation recommends it, as it is not the correct prescription. Can you show me a screenshot of that part?

**Comment by jorgecastro316** [comment]
![Screenshot from 2024-11-12 13-18-03](https://github.com/user-attachments/assets/24d676a3-cecd-441b-8169-635ef9bf50bf)

**Comment by mahrossi** [comment]
Do you think adding that BZP (first phrase) works with PIMD would make it clearer? and maybe also saying what Suzuki-Chin (higher order approx.) PIMD is?

**Comment by venkatkapil24** [comment]
Hi Jorge, I will edit the documentation to ensure there is no confusion. But returning to the original problem, do you also get these jumps in the pressure with the standard 'isotropic' barostat?

**Comment by jorgecastro316** [comment]
Yes, also in the standard isotropic setup. I've attached the input files. With these parameters, the simulation's got a bunch of outliers, but it still hits the target average pressure.
[press_pimd.zip](https://github.com/user-attachments/files/17728172/press_pimd.zip)

**Comment by venkatkapil24** [comment]
Hi Jorge, it would be nice if you could send a plot showing the outliers :)

**Comment by mahrossi** [comment]
I am running the calculations locally to check this.

**Comment by mahrossi** [comment]
I cannot reproduce it locally with the isotropic barostat and the inputs that were shared. I will try - just to see whether the finite diffs are to blame -- with the sc barostat. Maybe there is some thing going on with how the simulation is being run, but I am not sure.
